libcontainerd/supervisor/remote_daemon.go

```diff
@@ -187,7 +188,17 @@ func (r *remote) startContainerd() error {
 		args = append(args, "--log-level", r.logLevel)
 	}
 
-	cmd := exec.Command(binaryName, args...)
+	// In Windows containerd was first adopted by other tools so make sure that we
+	// use only containerd.exe which is provided with Docker binaries.
+	binPath := binaryName
+	if runtime.GOOS == "windows" {
+		dockerdPath, err := os.Executable()
+		if err == nil {
+			binPath = filepath.Join(filepath.Dir(dockerdPath), binaryName)
+		}
```

Comment:
We could change this to make this a "preferred" binary;

- look for containerd.exe in the same path as dockerd.exe; if found, use that
- otherwise use containerd.exe from `$PATH` (as before this patch)

Reply:
+1; I like that :+1:

Reply:
That approach is Raymond Chen approved, too! https://devblogs.microsoft.com/oldnewthing/20110620-00/?p=10393